modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/order/stock/test/OrderStockManagementTest.java

```diff
@@ -364,6 +365,54 @@ public void testProductReservedQuantity() throws Exception {
 			BigDecimal.valueOf(8));
 	}
 
+	@Test
+	public void testSubmitOrderWithoutInventory() {
+		frutillaRule.scenario(
+			"An order with not available product cannot be checked out"
+		).given(
+			"An order with a product with no stock availability"
+		).when(
+			"I try to checkout the order"
+		).then(
+			"An exception should be thrown"
+		);
+
+		try {
+			CommerceOrder commerceOrder = CommerceTestUtil.addB2CCommerceOrder(
+				_user.getUserId(), _commerceChannel.getGroupId(),
+				_commerceCurrency);
+
+			CommerceInventoryWarehouse commerceInventoryWarehouse =
+				CommerceInventoryTestUtil.addCommerceInventoryWarehouse(
+					ServiceContextTestUtil.getServiceContext(
+						_group.getGroupId()));
+
+			_commerceOrders.add(commerceOrder);
+
+			CPInstance cpInstance = CPTestUtil.addCPInstance(
+				_group.getGroupId());
+
+			CommerceInventoryTestUtil.addCommerceInventoryWarehouseItem(
+				_user.getUserId(), commerceInventoryWarehouse, BigDecimal.TEN,
+				cpInstance.getSku(), StringPool.BLANK);
+
+			CommerceTestUtil.addCommerceOrderItem(
+				commerceOrder.getCommerceOrderId(),
+				cpInstance.getCPInstanceId(), BigDecimal.valueOf(1));
+
+			CommerceInventoryTestUtil.addCommerceInventoryWarehouseItem(
+				_user.getUserId(), commerceInventoryWarehouse, BigDecimal.ZERO,
+				cpInstance.getSku(), StringPool.BLANK);
+
+			_commerceOrderEngine.checkoutCommerceOrder(
+				commerceOrder, _user.getUserId());
+		}
+		catch (Exception exception) {
+			Assert.assertSame(
+				CommerceOrderValidatorException.class, exception.getClass());
+		}
```

Comment:
Remove try/catch and add `@Test(expected = CommerceOrderValidatorException.class)`